apps/dashboard/app/views/files/index.html.erb

```diff
@@ -781,28 +781,35 @@ function reportTransfer(data){
   // 1. add the transfer label
   findAndUpdateTransferStatus(data);
 
+  let attempts = 0
+
   // 2. poll for the updates
-  var poll = function(){
-   $.getJSON(data.show_json_url, function(newdata){
-     findAndUpdateTransferStatus(newdata);
-
-     if(newdata.completed){
-       if(! newdata.error_message){
-         if(newdata.target_dir == history.state.currentDirectory){
-           reloadTable();
-         }
-
-         // 3. fade out after 5 seconds
-         fadeOutTransferStatus(newdata)
-       }
-     }
-     else{
-       // not completed yet, so poll again
-       setTimeout(poll, 1000);
-     }
-   }).fail(function(){
-     setTimeout(poll, 1000);
-   });
+  var poll = function() {
+    $.getJSON(data.show_json_url, function (newdata) {
+      findAndUpdateTransferStatus(newdata);
+
+      if(newdata.completed) {
+        if(! newdata.error_message) {
+          if(newdata.target_dir == history.state.currentDirectory) {
+            reloadTable();
+          }
+
+          // 3. fade out after 5 seconds
+          fadeOutTransferStatus(newdata)
+        }
+      }
+      else {
+        // not completed yet, so poll again
+        setTimeout(poll, 1000);
+      }
+    }).fail(function() {
+      if (attempts >= 3) {
+        Swal.fire('Operation may not have happened', 'Failed to retrieve file operation status.', 'error');
+      } else {
+        setTimeout(poll, 1000);
+        attempts++
```

Comment:
You're missing a semi-colon here. I'm guessing js linter doesn't recognize files with `.erb` extension? :cry: All the more reason to get to a world where javascript is javascript.

Reply:
@msquee we're still missing semi colons in a few places.